### DISCUSSION TITLE
USB Shell sample blank?

### DISCUSSION THREAD

**Discussion by paradajz** [zephyrproject-rtos/zephyr#73016] [General] [answered]
So the `main.c` file is blank, that is, it does nothing, therefore the USB shell sample doesn't work.

https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/usb/shell/src/main.c

Looking at the commit history there never was any code there. Is there a working sample of USB shell functionality?

**Comment by ndrs-pst** [comment, marked as answer]
@paradajz I’m not sure if you’ve already tried to build it or not.
Since the shell runs on a separate thread and initialization is done before reaching main, it’s common to see a blank `main.c`.

If it helps, you can check this link for another example of a blank `main.c`: 
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/shell/fs/src/main.c
